image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
+If you do choose to set a default password, ensure that an appropriate warning message is displayed when the container is started. The message should inform the user of the value of the default password and explain how to change it, such as what environment variable to set.
 
-If you do choose to set a default password, ensure that an appropriate warning message is displayed when the container is started.  The message should inform the user of the value of the default password and explaining how to change it (e.g. what environment variable to set).
+*Avoiding SSHD*
 
-=== Do not start (or install) SSHD
-Running SSHD should be avoided.  For accessing running containers you can use `docker exec` locally, or the OpenShift tooling which allows execution of arbitrary commands in running images.  Installing and running SSHD in your image opens up additional vectors for attack and requirements for security patching.
+Do not start or install SSHD in your image. For accessing running containers, you can use the `docker exec` command locally, or the OpenShift tooling which allows execution of arbitrary commands in running images. Installing and running SSHD in your image opens up additional vectors for attack and requirements for security patching.
 
-=== Use Volumes for Persisted Data
-Images should expect to use a https://docs.docker.com/reference/builder/#volume[Docker volume] for persisted data.  This way the OpenShift system can mount network storage to the node where the container is running and then reattach that storage at a new node if the container moves.  By using the volume for all persisted storage needs, the content will be preserved throughout container restarts and moves.  If your image writes data to arbitrary locations within the container, there is no guarantee that content will be preserved.
+*Using Volumes for Persisted Data*
 
-All data which is expected to live longer than the container must be written to a volume.  With Docker 1.5 there will be a `readonly` flag for containers which can be used to strictly enforce good practices about not writing data to ephemeral storage in a container, so designing your image around that capability now will make it easier to take advantage of it later.
+Images should expect to use a https://docs.docker.com/reference/builder/#volume[Docker volume] for persisted data. Doing so allows OpenShift to mount network storage to the node where the container is running, then reattach that storage at a new node if the container moves. By using the volume for all persisted storage needs, the content is preserved throughout container restarts and moves. If your image writes data to arbitrary locations within the container, there is no guarantee that content will be preserved.
```

Comment:
[Suggestion/Question]
"Images should expect to use a https://docs.docker.com/reference/builder/#volume[Docker volume] for persisted data." 

This statement doesn't make much sense to me. Should "images expect" or should a "user of images expect"? I'm not sure what we're trying to say here. Can we clarify this?

Doing so allows OpenShift to mount network storage to the node where the container is running, then reattach that storage at a new node if the container moves. ==>> This way OpenShift mounts the network storage to the node running the container, and if the container moves to a new node the storage is reattached to that node. By using the volume for all persisted storage needs, the content is preserved even if the container is restarted or moved. If your image writes data to arbitrary locations within the container, that content might not be preserved.